### DISCUSSION TITLE
How do I find out why a formula was removed?

### DISCUSSION THREAD

**Discussion by Mahoney** [Homebrew/discussions#5231] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.15
ORIGIN: https://github.com/Homebrew/brew
HEAD: 92a4311868322188478d7a90511ec0e8e6b0d7df
Last commit: 2 days ago
Core tap JSON: 26 Mar 12:30 UTC
Core cask tap JSON: 26 Mar 12:30 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.nPJ7xEebUT/org.xquartz:0
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.44.0 => /opt/homebrew/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.3
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  python-pathspec
```

### Description of issue

I've got `yamllint` installed, which depends on `python-pathspec`, which has been removed - see output of `brew doctor` above.

Consequently I cannot `brew uninstall python-pathspec`:
```
Refusing to uninstall /opt/homebrew/Cellar/python-pathspec/0.12.1
because it is required by yamllint, which is currently installed.
```

[`yamllint` documents using brew to install it](https://yamllint.readthedocs.io/en/stable/quickstart.html) so presumably they expect their formula to work, so I'd like to report it as an issue for them, but it would be nice to be able to explain _why_ their dependency has disappeared at the same time.

I tracked down [the PR #166358](https://github.com/Homebrew/homebrew-core/pull/166358) but it provides no rationale for the removal.

So my meta question is what's the right way to ensure people can work out why a formula was removed, so they can work out what (if anything) to do about it?

I found [Removal](https://docs.brew.sh/Deprecating-Disabling-and-Removing-Formulae#removal), but as far as I could see the formula hadn't been deprecated or disabled

**Comment by Mahoney** [comment]
And a `brew reinstall yamllint` removes the dependency so I can `brew uninstall python-pathspec`... that seems odd? I had done a `brew update && brew upgrade`, so why did I need to do a reinstall of a formula?

**Comment by p-linnane** [comment]
This is a special case where there is an ongoing project to stop using dedicated formulae in place of PyPI resources: https://github.com/Homebrew/homebrew-core/issues/157500

**Reply by Mahoney** [reply]
Thanks - wouldn't a link to that issue in the PR, allowing those knowing nothing about it to follow the reasoning back, have made sense?
